src/Domain/DomainCreators/BinaryCompactObject.hpp

```diff
@@ -0,0 +1,141 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+#include "Domain/Domain.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <size_t, class>
+class DomainCreator;  // IWYU pragma: keep
+/// \endcond
+
+namespace DomainCreators {
+
+/*!
+ * \ingroup DomainCreatorsGroup
+ *
+ * \brief Creates a 3D Domain consisting of two hemispheres of blocks put
+ * together, where each hemisphere has four layers of blocks:
+ * - The first layer has the blocks for the compact object.
+ * - The second layer has the blocks that surround the object with a cube.
+ * - The third layer has the blocks that surround each cube with a half-cube.
+ * - The fourth layer is made of the blocks that surround the enveloping cube
```

Comment:
@kidder 
@nilsdeppe 

I am thinking that I would like a 4th, generic binary compact object domain creator whose sole purpose is to be used for testing the implementation details by being fully customizable in terms of the domain options. There will be no physical options passed to this domain, only options for the sizes and maps of the different elements of the domain.

I have been wondering, given a initial specification of options to say the BBH domain, is the domain further optimized to accomplish a good numerical simulation?

What I am seeing is there are two ways a user can think about providing the options to the domain creator - the first is by specifying the physical options - the second is by specifying the coordinate sizes and properties of each of the elements of the domain. Given that a user would want the "best" domain for a given physical system, how are the parameters for the element sizes/maps determined? I assume the control system takes care of a lot of this dynamically, but what about for the initial state of the domain? I assume this was also something that had to be addressed in SpEC, is there a paper or discussion about this particular issue? Given that we are also going to be a using a domain decomposition that is different from that of SpEC's, I want to be confident that the domain constructed from a user's specification of the physical system is indeed the "best" (under some measure) suited for simulating that physical system. (This is also why I would like the 4th, generic BCO domain creator where the user can override this for testing purposes, maybe even for testing the control system to see if it can take two different domain specifications for the same physical system and bring them close to some third, "ideal" domain)

Reply:
This sounds like a good idea.  The SpEC domains were made pretty much by experimentation, so it would be good to have the generic experimental domain creator first.  See https://arxiv.org/pdf/1405.3693.pdf for some discussion of the SpEC BBH domain